web/src/HUD.tsx

```diff
@@ -190,9 +194,25 @@ class HUD extends Component<HudProps, HudState> {
               snapshotLink: value.url ? value.url : "",
             })
           })
-          .catch(err => console.error(err))
+          .catch(err => {
+            console.error(err)
+            if (err instanceof Error) {
+              this.setAppState({
+                showSnapshotModal: false,
+                error: `Error sending snapshot: ${err.message}`,
+              })
+            } else {
+              this.setAppState({
+                showSnapshotModal: false,
+                error:
+                  "Error sending snapshot. Check JavaScript console for more details",
```

Comment:
@jazzdan : I don't think we should be asking a user to check the browser console for more details. If it's important for the user to see the details, can't we surface that information inline in the error here? I don't think we should be asking the user to debug the problem for us.

Reply:
Yeah I'm not an expert in JS error handling but it seems like sometimes this error value isn't very helpful. Maybe I should do a `try`/`catch` too?

Reply:
@jazzdan : I was just trying to say I don't think we should show the words "Check JavaScript console for more details" to the user in their regular error flow. If that doesn't actually appear to the user, then it's fine. But if it does appear, I think we should remove it. 

I understand our users our technical and would certainly understand the message. But I don't think we should be exposing this in the main UX itself.

So simply:

```suggestion
                  "Error sending snapshot.",
```

Don't know what you mean by try/catch here. You'd have to ask someone else!